src/util/strencodings.cpp

```diff
@@ -201,20 +201,24 @@ std::string DecodeBase64(const std::string& str, bool* pf_invalid)
     return std::string((const char*)vchRet.data(), vchRet.size());
 }
 
-std::string EncodeBase32(Span<const unsigned char> input)
+std::string EncodeBase32(Span<const unsigned char> input, bool pad)
 {
     static const char *pbase32 = "abcdefghijklmnopqrstuvwxyz234567";
 
     std::string str;
     str.reserve(((input.size() + 4) / 5) * 8);
     ConvertBits<8, 5, true>([&](int v) { str += pbase32[v]; }, input.begin(), input.end());
-    while (str.size() % 8) str += '=';
+    if (pad) {
+        while (str.size() % 8) {
+            str += '=';
+        }
+    }
```

Comment:
1046e17c would be good to add an explanation of why `pad` is needed, maybe in a doxygen doc for the function in the header file `strencodings.h`

Reply:
I don't know why the padding is needed! :) But I added a comment on what the newly added parameter does.

Reply:
Thanks.